libcontainer/container_linux.go

```diff
@@ -276,6 +276,21 @@ func (c *linuxContainer) Exec() error {
 func (c *linuxContainer) exec() error {
 	path := filepath.Join(c.root, execFifoFilename)
 
+	defer func() {
+		if c.config.Hooks != nil && len(c.config.Hooks.Poststart) > 0 {
```

Comment:
I don't know how cheap goroutines are, but you may want:

```go
if c.config.Hooks != nil && len(c.config.Hooks.Poststart) > 0 {
  defer func(0 {
    ...
  }()
}
```

To avoid spawning one when there are no hooks.